src/TChar.cpp

```diff
@@ -0,0 +1,61 @@
+/***************************************************************************
+ *   Copyright (C) 2008-2013 by Heiko Koehn - KoehnHeiko@googlemail.com    *
+ *   Copyright (C) 2014 by Ahmed Charles - acharles@outlook.com            *
+ *   Copyright (C) 2014-2018 by Stephen Lyons - slysven@virginmedia.com    *
```

Comment:
Please :pleading_face: add your details to files that you modify - whilst it is possible to identify contributors from the repository history - that doesn't actually work as far as Copyright law actually goes (AFAICT) - and for new source code files please insert our standard GPL 2+ boilerplate (see the `./templates/README` for a hint on how to do that automagically within Qt Creator using the other files in that directory) to all of them!